docs/guides/middleware.md

```diff
@@ -104,45 +135,78 @@ var myMiddleware = function(player) {
     setCurrentTime: function(time) {
       return time * 2;
     },
-    ...
+    setSource: function(srcObj, next) {
+      // pass null as the first argument to indicate that the source is not rejected
+      next(null, srcObj);
+    }
   };
 };
 
 videojs.use('*', myMiddleware);
 ```
 
-### Terminating Mediator Methods
+This middleware gives the appearance of the video source playing at double its speed, by halving the time we _get_ from the `Tech`, and doubling the time we _set_ on the `Tech`.
 
-Mediator methods can terminate, by doing the following:
+An example of a middleware that uses Mediator methods is below:
 
 ```javascript
 var myMiddleware = function(player) {
   return {
     callPlay: function() {
-      // Terminate by returning the middleware terminator
-      return videojs.middleware.TERMINATOR;
+      // Do nothing, thereby allowing play() to be called on the Tech
     },
     play: function(terminated, value) {
-      // the terminated argument should be true here.
+      if (terminated) {
+        console.log('The play was middleware terminated.');
+
+      // the value is a play promise
+      } else if (value && value.then) {
+        value
+          .then(function() {
+            console.log('The play succeeded!')
+          })
+          .catch(function (err) {
+            console.log('The play was rejected', err);
+          });
+      }
     },
-    ...
+    setSource: function(srcObj, next) {
```

Comment:
can we move `setSource` to be the first property of the object? Since it's required and it's important to know which source it's selected with, having it first, I think, is useful.

Reply:
below as well